packages/devtools_app/test/network/network_request_inspector_test.dart

```diff
@@ -129,5 +130,197 @@ void main() {
+        final requestsNotifier = controller.requests;
+        await controller.startRecording();
+
+        // Load the network request.
+        await controller.networkService.refreshNetworkData();
+        expect(requestsNotifier.value.requests.length, equals(1));
+
+        // Select the request in the network request list.
+        final data = requestsNotifier.value.requests.first;
+
+        await tester.pumpWidget(
+          wrapWithControllers(
+            HttpResponseTrailingDropDown(
+              data as DartIOHttpRequestData,
+              currentResponseViewType: currentResponseViewType,
+              onChanged: (value) {
+                controller.setResponseViewType = value;
+              },
+            ),
+            debugger: createMockDebuggerControllerWithDefaults(),
+          ),
+        );
+
+        final dropDownFinder = find.byType(
+          RoundedDropDownButton<NetworkResponseViewType>,
+        );
+
+        await tester.tap(dropDownFinder);
+        await tester.pumpAndSettle();
+
+        // Select Json from drop down
+        await tester.tap(
+          find.text(
+            NetworkResponseViewType.json.toString(),
+          ),
+        );
+        await tester.pumpAndSettle();
+        expect(
+          currentResponseViewType.value,
+          equals(NetworkResponseViewType.json),
+        );
+
+        await tester.tap(dropDownFinder);
+        await tester.pumpAndSettle();
+
+        // Select Text from drop down
+        await tester.tap(
+          find.text(
+            NetworkResponseViewType.text.toString(),
+          ),
+        );
+        await tester.pumpAndSettle();
+        expect(
+          currentResponseViewType.value,
+          equals(NetworkResponseViewType.text),
+        );
+
+        await tester.tap(dropDownFinder);
+        await tester.pumpAndSettle();
+
+        // Select Auto from drop down
+        await tester.tap(
+          find.text(
+            NetworkResponseViewType.auto.toString(),
+          ),
+        );
+        await tester.pumpAndSettle();
+        expect(
+          currentResponseViewType.value,
+          equals(NetworkResponseViewType.auto),
+        );
+
+        controller.stopRecording();
+
+        // pumpAndSettle so residual http timers can clear.
+        await tester.pumpAndSettle(const Duration(seconds: 1));
+      });
+    });
+
+    testWidgets(
+        'should update response view display when drop down value changes',
+        (tester) async {
+      final requestsNotifier = controller.requests;
+      await controller.startRecording();
+
+      await tester.pumpWidget(
+        wrapWithControllers(
+          NetworkRequestInspector(controller),
+          debugger: createMockDebuggerControllerWithDefaults(),
+        ),
+      );
+
+      // Load the network request.
+      await controller.networkService.refreshNetworkData();
+      expect(requestsNotifier.value.requests.length, equals(1));
```

Comment:
no need to interact with the controllers at all here either. How you know whether this is needed is look at the `HttpTextResponseViewer` widget directly and see if it ever interacts with controllers or if one is passed in as a parameter. In this case it is not, so we can just pump the `HttpTextResponseViewer` widget directly with the parameter values we want to test with.

This is also the answer to your question:
> How can I change responseBody or contentType to test my widgets against that? I need to have different kinds of HTTP response to test that. Currently I just take one request from the requests list coming from `NetworkController`

Just pass in the values for `contentType` and `responseBody` that you want directly when you pump the widget.